apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1203,9 +1206,13 @@ def _write_dataframe_impl(
         arrow_table = _extract_new_values_for_append(df_uri, arrow_table, context)
 
     try:
+        domain = None
+        if NEW_SHAPE_FEATURE_FLAG_ENABLED:
+            domain = ((0, shape - 1),)
```

Comment:
It looks like this is now assuming the dataframe uses the `soma_joinid` as the one and only index column. If this is the correct assumption (and I think it probably is for the anndata ingestor), can you add a comment about that in the docstring?

Reply:
I'll do so

Yes `tiledbsoma.io` does choose `soma_joinid` as the one and only index column.